File: grading-students.js
function gradingStudents(grades) {
  grades = grades.map((grade) => {
    if (grade < 38) return grade;
    let digit = Number((grade + '')[1]);
    if (digit === 9 || digit === 4) return grade + 1;
    if (digit === 8 || digit === 3) return grade + 2;
    return grade;
  });
  return console.log(grades);
}

let grades = [73, 67, 38, 33];

gradingStudents(grades);

File: grading-students.kt
fun gradingStudents(grades: Array<Int>): Array<Int> {
  var list = mutableListOf<Int>()
  grades.forEach {
    val mod = it % 5
    val num = if (it > 37 && mod >= 3) it + (5 - mod) else it
    list.add(num)
  }
  return list.toTypedArray()
}